### ISSUE TITLE
Helm3 upgrade --install does not respect the --version flag

### ISSUE THREAD

**Issue by lamLeX** [helm/helm#6820]
I used the 
```
helm upgrade --install $AppName $AppChartLocation --namespace=$AppNamespace --version=$AppChartVersion
```
to upgrade/install the chart. And 
`AppChartVersion=0.0.0-${CI_COMMIT_REF_SLUG}+${CI_COMMIT_SHORT_SHA}`

e.g. `0.0.0-7-featureA+47f46c46`, `0.0.0-7-featureA+959728d8`

In my chart repo, the `0.0.0-7-featureA+959728d8` is uploaded first and then `0.0.0-7-featureA+47f46c46`.

When I ran `helm upgrade --version=$AppChartVersion`(abbreviated for brevity) with `AppChartVersion=0.0.0-7-featureA+47f46c46`, helm installed the chart `0.0.0-7-featureA+959728d8`. I expected helm to respect the version that I provided which is `0.0.0-7-featureA+47f46c46` instead. 

Output of `helm version`:
`version.BuildInfo{Version:"v3.0.0-beta.5", GitCommit:"509a8819b7e8dba86b7982c006fce22628aeea28", GitTreeState:"clean", GoVersion:"go1.13.3"}`

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.10-eks-825e5d", GitCommit:"825e5de08cb05714f9b224cd6c47d9514df1d1a7", GitTreeState:"clean", BuildDate:"2019-08-18T03:58:32Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
EKS

**Comment by bacongobbler** [comment]
I think this may be a bug where a fix available in Helm 2 was not ported over to Helm 3. Feel like working on porting over a fix?

**Comment by bacongobbler** [comment]
See https://github.com/helm/helm/pull/4850.

**Comment by bacongobbler** [comment]
fixed in #6821
